src/modules/mod.ts

```diff
@@ -20,4 +23,30 @@ export class ModModule extends Module {
 			`${msg.author} We don't do job posts here; see <#${rulesChannelId}>`,
 		);
 	}
+
+	@listener({ event: 'guildMemberAdd' })
+	async onJoin(member: GuildMember) {
+		const now = Date.now();
+		const lowerBound = now - RAID_SECONDS * 1000;
+
+		this.joins.push(member);
+
+		// Clean up joins from from outside of our raid threshold.
+		while ((this.joins[0].joinedTimestamp ?? 0) < lowerBound) {
+			this.joins.splice(0, 1);
+			this.bannedUpTo--;
+		}
+
+		// Ban everyone that just joined
+		if (this.joins.length > RAID_JOINS) {
+			const start = this.bannedUpTo;
+			this.bannedUpTo = this.joins.length;
+
+			await Promise.all(
+				this.joins
+					.slice(start)
```

Comment:
For the record, the negative number in `bannedUpTo` might lead to things like `.slice(-1)` here, which might mean that we detect raids properly, but don't ban the first `RAID_JOINS - 1` accounts.